### ISSUE TITLE
MaxMemory behavior is different from other frameworks

### ISSUE THREAD

**Issue by a631807682** [astaxie/beego#4054]
Please answer these questions before submitting your issue. Thanks!

1. What version of Go and beego are you using (`bee version`)?
 1.12.2

2. What operating system and processor architecture are you using (`go env`)?
 1.14.2

3. What did you do?
Post with a body size larger then `MaxMemory`.

4. What did you expect to see?
Response http code 413.

5. What did you see instead?
A truncated body is passed.

6. More detail
`MaxMemory` behavior is different from other the framework that I've used, and it doesn't seem to satisfy the [rfc7231#section-6.5.11](https://tools.ietf.org/html/rfc7231#section-6.5.11)
I think this code is problematic [router.go#L743-L748](https://github.com/astaxie/beego/blob/develop/router.go#L743-L748)
```
	if r.Method != http.MethodGet && r.Method != http.MethodHead {
		if BConfig.CopyRequestBody && !context.Input.IsUpload() {
			context.Input.CopyBody(BConfig.MaxMemory)
		}
		context.Input.ParseFormOrMulitForm(BConfig.MaxMemory)
	}
```
We should response error code rather than pass a truncated body. Silent errors make it difficult to troubleshoot errors.
Here is my test
```
// copy to router_test.go and run this test
func TestRouterPostLargeBody(t *testing.T) {
	maxMemory := BConfig.MaxMemory
	BConfig.CopyRequestBody = true
	BConfig.MaxMemory = 20

	type reqBody struct {
		Foo string `json:"foo"`
	}

	originBody := &reqBody{
		Foo: "barbarbarbarbar",
	}

	s, _ := json.Marshal(originBody)
	b := bytes.NewBuffer(s)
	r, _ := http.NewRequest("POST", "/user/123", b)
	w := httptest.NewRecorder()

	handler := NewControllerRegister()
	handler.Post("/user/:id", func(ctx *context.Context) {
		var body reqBody
		if err := json.Unmarshal(ctx.Input.RequestBody, &body); err != nil {
                       // TestRouterPostLargeBody unmarshal body size, recive {"foo":"barbarbarbar
			t.Errorf("TestRouterPostLargeBody unmarshal body size, recive %s", string(ctx.Input.RequestBody))
		}

		if originBody.Foo != body.Foo {
			t.Errorf("TestRouterPostLargeBody request and recive body not equal")
		}
		BConfig.MaxMemory = maxMemory

		var res []byte
		ctx.Output.Body(res)
	})
	handler.ServeHTTP(w, r)
}

```

**Comment by a631807682** [comment]
简单的说就是当我们在使用`MaxMemory`时，我们希望大于`MaxMemory`直接返回`413`，而不是静默的截取`body`并透传下去。
因为截取的`RequestBody`通常没有意义，还会使错误更混乱，特别是我们可能在`controller`中作代理别的`http`请求去透传`RequestBody`的时候。

**Comment by flycash** [comment]
You are right. We can fix this in 2.0.0 because this is a broken change. Could you please help to fix it?

**Comment by a631807682** [comment]
I will create a PR this week.
